### ISSUE TITLE
Cannot Push to HEAD

### ISSUE THREAD

**Issue by lssilva** [uwolfer/gerrit-intellij-plugin#175]
I could not find a way to push to HEAD:refs/for/master it only offers the option to push to :refs/for/master

**Comment by uwolfer** [comment]
@lssilva: Why do you need this option? Does :refs/for/master not work for you?

**Comment by Suavac** [comment]
@uwolfer, I'm getting the 'Push Rejected' message for the reason mentioned by @lssilva, so it forces me to use the git bash for 'push origin HEAD:refs/for/master'.

**Comment by uwolfer** [comment]
Can you please provide the full (anonymized) error which you can find in IntelliJ "Version Control"-tab log?

**Comment by uwolfer** [comment]
@Suavac / @lssilva: Please see my question above. Any feedback would be appreciated. Thanks.

**Comment by lssilva** [comment]
only refs won't work for my current project I need to push 
git push origin HEAD:refs/for/rel-2.3.X
git push origin HEAD:refs/for/rmaster

That is just how the project is setup and I have no access to change it.

**Comment by uwolfer** [comment]
@lssilva: Can you please provide the full (anonymized) error which you can find in IntelliJ "Version Control"-tab log?

**Comment by lssilva** [comment]
I don't think there is a error it is just not possible to add HEAD anywhere

**Comment by uwolfer** [comment]
@lssilva: Ah, I thought you get the same error as @Suavac when you try to push to refs/for/... (without HEAD).

**Comment by lssilva** [comment]
http://stackoverflow.com/questions/10461214/why-is-git-push-gerrit-headrefs-for-master-used-instead-of-git-push-origin-mast
A more detailed discussion here

**Comment by uwolfer** [comment]
@lssilva: Sure, "refs/for/.." is required in order to push to Gerrit. I'm just not sure why "HEAD" is absolutely required in your case.

**Comment by uwolfer** [comment]
@Suavac: Can you please provide the full (anonymized) error which you can find in IntelliJ "Version Control"-tab "log" when you get the "push rejected" error?

**Comment by lssilva** [comment]
I just tried to do it to get you a log. In the git push view I replaced master with my URL 

HEAD:refs/for/master 
and I got a pop-up Invalid destination branch name: HEAD:refs/for/master I have a screenshot but cannot attach here.

**Comment by uwolfer** [comment]
@lssilva: You can attach screenshots here - just drag it into the comment field.

**Comment by lssilva** [comment]
![screen shot 2015-11-16 at 14 03 18](https://cloud.githubusercontent.com/assets/7464888/11206342/b8ccd2a2-8d0e-11e5-8d34-62146567f56a.png)

**Comment by uwolfer** [comment]
@lssilva: What version of IntelliJ are you using?

Do you have enabled the option "Push to Gerrit" in the push dialog? When exactly does this message pop up?

**Comment by nsharma-git** [comment]
I see same issue. All I can see in IntelliJ log is - 
**2:43:29 PM Push rejected: Push to origin/refs/for/master was rejected**

I"m using Gerrit 2.12.2 and Plugin version 0.9.8.1-142.
As stated in this thread, I can push the changes to gerrit for review as git push origin HEAD:refs/for/master

**Comment by tunix** [comment]
I'm having the same issue with Gerrit 2.12 and Plugin version 0.9.9.1-142.

@uwolfer I found out that I'm getting this error initially for the lack of change-id in my commits. Is there a way for the plugin to add change id by automatically adding the hook and amending commits? (asking because git review CLI tool is able to that)

After manually adding the change id, it worked for me.

**Comment by uwolfer** [comment]
@tunix: No, it is not possible to add change ids after committing automatically with the plugin. If you'd like this feature, you can open a feature request for it.

**Comment by dadpem** [comment]
Same thing with me. I want to add HEAD as HEAD:refs/for/"Branch Name", but it doesn't allow me to add and I can not push my commit
![image](https://cloud.githubusercontent.com/assets/12809369/23744238/6d84e1c4-04d9-11e7-8553-03b829f333bb.png)

**Comment by Etesyolk** [comment]
Hi, please somebody explain me how to use this plugin properly.
I have a Linux machine with installed Gerrit on it and i want to manage my code review from Intelij IDE, which runs on the local Windows machine. I have already gerrit plugin installed on my intelij and i connected to gerrit server, but when i'm pushing the commits with gerrit options, the commits are never getting the gerrit server, despite i'm getting sometimes the message "pushed successfully". What is wrong?

**Comment by uwolfer** [comment]
@Etesyolk: You can see details about your push in `Version Control -> Log` tool window. If this still does not help, please open a new issue here on GitHub.
